projects/plugins/jetpack/modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-universal.php

```diff
@@ -625,6 +625,48 @@ public static function get_cart_checkout_info() {
 				),
 			);
 
+			// If WC Blocks is version 10.6.0 or above, it's using the Cart/Checkout templates.
+			$cart_checkout_templates_in_use = class_exists( 'Automattic\WooCommerce\Blocks\Package' ) && version_compare( Automattic\WooCommerce\Blocks\Package::get_version(), '10.6.0', '>=' );
+
+			$cart_template        = null;
+			$checkout_template    = null;
+			$cart_template_id     = null;
+			$checkout_template_id = null;
+
+			if ( $cart_checkout_templates_in_use ) {
+				$block_controller = Automattic\WooCommerce\Blocks\Package::container()->get( Automattic\WooCommerce\Blocks\BlockTemplatesController::class );
+				$templates        = $block_controller->get_block_templates( array( 'cart', 'checkout' ) );
+
+				foreach ( $templates as $template ) {
+					if ( 'cart' === $template->slug ) {
+						$cart_template_id = ( $template->id );
+						continue;
+					}
+					if ( 'checkout' === $template->slug ) {
+						$checkout_template_id = ( $template->id );
+					}
+				}
+
+				// Get the template and its contents from the IDs we found above.
+				if ( function_exists( 'get_block_template' ) ) {
+					$cart_template     = get_block_template( $cart_template_id );
+					$checkout_template = get_block_template( $checkout_template_id );
+				}
+
+				if ( function_exists( 'gutenberg_get_block_template' ) ) {
+					$cart_template     = get_block_template( $cart_template_id );
+					$checkout_template = get_block_template( $checkout_template_id );
+				}
+
+				// Update the info transient with data we got from the templates, if the site isn't using WC Blocks we
+				// won't be doing this so no concern about overwriting.
+				$info = array(
+					'cart_page_contains_cart_block'     => str_contains( $cart_template->content, '<!-- wp:woocommerce/cart' ),
+					'cart_page_contains_cart_shortcode' => str_contains( $cart_template->content, '[woocommerce_cart]' ),
+					'checkout_page_contains_cart_block' => str_contains( $checkout_template->content, '<!-- wp:woocommerce/checkout' ),
```

Comment:
Hm, this should not trigger an error since WordPress includes a polyfill. @anomiex Do you know why this is happening?

Reply:
> For now, adding to .github/files/phpcompatibility-dev-phpcs.xml is probably the most sensible thing.

Is this something I should add in this PR? Happy to do so just need to confirm it's the right step.